### PR TITLE
Refactor node agent.

### PR DESCRIPTION
**Pull request by myidpt**
**Release note**:

```release-note
* Add secret server interface to support multiple interfaces to propagate secretes to Envoy.
* Add FileUtil interface to support mocking file operations in tests.
* Add CertUtil interface to support mocking in tests.
* Add tests for node agent, etc.
```

### CODE REVIEW

File: pkg/workload/secretserver.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	// SetServiceIdentityPrivateKey sets the service identity private key to the channel accessible to the workload.
+	SetServiceIdentityPrivateKey([]byte) error
+	// SetServiceIdentityCert sets the service identity cert to the channel accessible to the workload.
+	SetServiceIdentityCert([]byte) error
+}
+
+// NewSecretServer instantiates a SecretServer according to the configuration.
+func NewSecretServer(cfg Config) (SecretServer, error) {
+	switch cfg.Mode {
+	case SECRETFILE:
```
Comment (wattli) [inline pkg/workload/secretserver.go]: Put a underscore in between?

File: pkg/workload/secretserver.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		return &SecretFile{cfg}, nil
+	case WORKLOADAPI:
+		return nil, fmt.Errorf("WORKLOAD API is unimplemented")
+	default:
+		return nil, fmt.Errorf("Mode is not supported: %d", cfg.Mode)
```
Comment (wattli) [inline pkg/workload/secretserver.go]: How about "Mode: %d is not supported."?

**Comment by wattli** [comment]
Thanks for putting this up. Can we add a little bit more info to explain what this PR actually does? This will make it easier to track in the future.

File: cmd/node_agent/na/util_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	"io/ioutil"
+	"testing"
+	"time"
+)
+
+const (
+	// This cert has:
+	//   NotBefore = 2017-08-23 19:00:40 +0000 UTC
+	//   NotAfter  = 2017-08-24 19:00:40 +0000 UTC
+	testCertPath = "testdata/cert-util.pem"
```
Comment (wattli) [inline cmd/node_agent/na/util_test.go]: testCertFile or testCertPathFile?

Comment (myidpt) [reply cmd/node_agent/na/util_test.go]: I meant testCertFile.

File: pkg/workload/secretserver.go (hunk 2 above)
Comment (myidpt) [reply pkg/workload/secretserver.go]: Sure.

File: pkg/workload/secretserver.go (hunk 1 above)
Comment (myidpt) [reply pkg/workload/secretserver.go]: Sure.

Comment (myidpt) [reply pkg/workload/secretserver.go]: Made them camel cases as required by the linter.

**Pull request merged**